<a href="https://colab.research.google.com/github/Florani1/Hamoye-Internship-Codes/blob/main/Tag_along_Flora_Akanbi.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Stage A Quiz

In [2]:
# import all packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
df =pd.read_csv("/content/FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding = "latin-1")

In [4]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [5]:
df.shape

(60943, 12)

In [27]:
delf = df.groupby('Element')['Y2014'].sum()
delf

Element
Domestic supply quantity                  1996716.35
Export Quantity                            150020.64
Fat supply quantity (g/capita/day)          10225.56
Feed                                       216927.89
Food                                      1212332.49
Food supply (kcal/capita/day)              454257.00
Food supply quantity (kg/capita/yr)         49650.63
Import Quantity                            274144.48
Losses                                     153223.00
Other uses (non-food)                       78718.13
Processing                                 282923.00
Production                                1931287.75
Protein supply quantity (g/capita/day)      11836.46
Residuals                                   30149.00
Seed                                        21922.92
Stock Variation                             58749.83
Total Population - Both sexes             1031585.00
Tourist consumption                           416.00
Name: Y2014, dtype: float64

7. 2014 has the highest stock variation

In [26]:
delf = df.groupby('Element')['Y2015'].sum()
delf

Element
Domestic supply quantity                  2021493.55
Export Quantity                            157614.47
Fat supply quantity (g/capita/day)          10235.74
Feed                                       225050.22
Food                                      1232361.10
Food supply (kcal/capita/day)              453383.00
Food supply quantity (kg/capita/yr)         49345.13
Import Quantity                            267018.46
Losses                                     155439.00
Other uses (non-food)                       66254.41
Processing                                 287929.00
Production                                1947019.39
Protein supply quantity (g/capita/day)      11833.95
Residuals                                   30045.00
Seed                                        23976.82
Stock Variation                             34910.99
Total Population - Both sexes             1058081.00
Tourist consumption                           349.00
Name: Y2015, dtype: float64

In [28]:
delf = df.groupby('Element')['Y2016'].sum()
delf

Element
Domestic supply quantity                  2044842.70
Export Quantity                            151920.46
Fat supply quantity (g/capita/day)          10102.77
Feed                                       228958.65
Food                                      1247022.17
Food supply (kcal/capita/day)              451810.00
Food supply quantity (kg/capita/yr)         48985.28
Import Quantity                            286582.78
Losses                                     157787.00
Other uses (non-food)                       69563.68
Processing                                 280631.00
Production                                1943537.15
Protein supply quantity (g/capita/day)      11779.69
Residuals                                   37224.00
Seed                                        23389.20
Stock Variation                             33140.12
Total Population - Both sexes             1085107.00
Tourist consumption                            89.00
Name: Y2016, dtype: float64

In [29]:
delf = df.groupby('Element')['Y2017'].sum()
delf

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

In [30]:
delf = df.groupby('Element')['Y2018'].sum()
delf

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

In [6]:
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

3. the mean for 2017 - 140.92
standard dev = 1671.86

1. Year 2014 has the least corelation with Element Code

In [24]:
df.describe().Y2014

count     59354.000000
mean        134.196282
std        1567.663696
min       -1796.000000
25%           0.000000
50%           0.090000
75%           8.340000
max      176405.000000
Name: Y2014, dtype: float64

In [7]:
df.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


TypeError: 'NoneType' object is not subscriptable

In [17]:
df.groupby['Element Code']['Y2014','Y2015','Y2016','Y2017','Y2018'].sum()

TypeError: 'method' object is not subscriptable

In [22]:
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

3. The total number and percentage of missing data in 2014 is 1589 and 2.608%

In [31]:
elf = df.groupby('Area').sum()
elf

Area Code  Item Code  \
Area                                                
Algeria                           5252    3532053   
Angola                            9968    3819908   
Benin                            68105    3453692   
Botswana                         27560    3704902   
Burkina Faso                    328297    3778795   
Cabo Verde                       44100    3396746   
Cameroon                         46720    3916078   
Central African Republic         49728    3607546   
Chad                             52377    3606529   
Comoros                           2025     124514   
Congo                            64998    3792029   
Côte d'Ivoire                   156969    3934818   
Djibouti                         78840    2956078   
Egypt                            87792    3996567   
Eswatini                        268356    3456174   
Ethiopia                        344386    3878162   
Ethiopia PDR                      2418     107853   
Gabon                            92130    3354410   
Gambia                           87600    3145402   
Ghana                           109674    3639339   
Guinea                          126900    3782145   
Guinea-Bissau                   197400    3035574   
Kenya                           177840    4185637   
Lesotho                         140300    3099348   
Liberia                         148338    3249929   
Madagascar                      185631    3864258   
Malawi                          188110    3879389   
Mali                            171969    3469876   
Mauritania                      162520    3222107   
Mauritius                       188786    3711587   
Morocco                         199199    3743082   
Mozambique                      205488    3830309   
Namibia                         188601    3452725   
Niger                           221674    3763737   
Nigeria                         234366    3954357   
Rwanda                          262384    3824085   
Sao Tome and Principe           227161    3168050   
Senegal                         285090    3922180   
Seychelles                       13328     188266   
Sierra Leone                    244674    3343306   
South Africa                    282598    3759379   
Sudan                           363216    3534661   
Sudan (former)                   10918     146631   
Togo                            280798    3478828   
Tunisia                         297036    3602818   
Uganda                          329508    3909109   
United Republic of Tanzania     305085    3809549   
Zambia                          364201    3892801   
Zimbabwe                        252133    3739292   

                                                                          Item  \
Area                                                                             
Algeria                      PopulationPopulationGrand TotalGrand TotalGran...   
Angola                       PopulationPopulationGrand TotalGrand TotalGran...   
Benin                        PopulationPopulationGrand TotalGrand TotalGran...   
Botswana                     PopulationPopulationGrand TotalGrand TotalGran...   
Burkina Faso                 PopulationPopulationGrand TotalGrand TotalGran...   
Cabo Verde                   PopulationPopulationGrand TotalGrand TotalGran...   
Cameroon                     PopulationPopulationGrand TotalGrand TotalGran...   
Central African Republic     PopulationPopulationGrand TotalGrand TotalGran...   
Chad                         PopulationPopulationGrand TotalGrand TotalGran...   
Comoros                      Fish, Body OilFish, Body OilFish, Body OilFish...   
Congo                        PopulationPopulationGrand TotalGrand TotalGran...   
Côte d'Ivoire                PopulationPopulationGrand TotalGrand TotalGran...   
Djibouti                     PopulationPopulationGrand TotalGrand TotalGran...   
Egypt                        PopulationPopulationGrand TotalGrand TotalGran...   
Eswatini     

In [38]:
elf = df.groupby('Area')['Y2017'].count()
elf

Area
Algeria                        1279
Angola                         1407
Benin                          1253
Botswana                       1368
Burkina Faso                   1383
Cabo Verde                     1193
Cameroon                       1441
Central African Republic       1262
Chad                           1281
Comoros                          45
Congo                          1395
Côte d'Ivoire                  1454
Djibouti                       1061
Egypt                          1458
Eswatini                       1251
Ethiopia                       1428
Ethiopia PDR                     39
Gabon                          1206
Gambia                         1116
Ghana                          1333
Guinea                         1383
Guinea-Bissau                  1076
Kenya                          1548
Lesotho                        1110
Liberia                        1135
Madagascar                     1425
Malawi                         1411
Mali                   

In [41]:
elf = df.groupby('Area')['Y2017'].sum()
elf

Area
Algeria                         325644.27
Angola                          229159.57
Benin                           124771.22
Botswana                         22101.30
Burkina Faso                    101855.07
Cabo Verde                       14650.74
Cameroon                        232030.43
Central African Republic         29937.00
Chad                             71594.68
Comoros                             59.84
Congo                            41181.68
Côte d'Ivoire                   224599.01
Djibouti                         22729.91
Egypt                           866379.92
Eswatini                         54343.33
Ethiopia                        448683.76
Ethiopia PDR                         0.00
Gabon                            27979.64
Gambia                           23154.18
Ghana                           337599.06
Guinea                           98138.87
Guinea-Bissau                    19102.77
Kenya                           264660.66
Lesotho                      